processor/k8sattributesprocessor/internal/kube/client.go

```diff
@@ -643,16 +661,14 @@ func (c *WatchClient) extractPodContainersAttributes(pod *api_v1.Pod) PodContain
 	if c.Rules.ContainerImageName || c.Rules.ContainerImageTag {
 		for _, spec := range append(pod.Spec.Containers, pod.Spec.InitContainers...) {
 			container := &Container{}
-			nameTagSep := strings.LastIndex(spec.Image, ":")
-			if c.Rules.ContainerImageName {
-				if nameTagSep > 0 {
-					container.ImageName = spec.Image[:nameTagSep]
-				} else {
-					container.ImageName = spec.Image
+			name, tag, err := parseNameAndTagFromImage(spec.Image)
+			if err == nil {
+				if c.Rules.ContainerImageName {
+					container.ImageName = name
+				}
+				if c.Rules.ContainerImageTag {
```

Comment:
👍🏽 

Since we are on it, I also wonder if we should override the `ImageTag` from the container's status in case it is available.
See https://github.com/kubernetes/api/blob/v0.32.0-alpha.3/core/v1/types.go#L3047-L3055.

Reply:
sorry, don't think I understand this point? i don't see any field `ImageTag` in `ContainerStatus`?

Reply:
I was referring to [`Image`](https://github.com/kubernetes/api/blob/dc555a599a635e3247477cb88d7ecdc5727311e8/core/v1/types.go#L3051) but checking it again it can also take a `sha:digest` as value so this would complex things. We could evaluate this in the future but let's leave it out for now.